spatial/r3/triangle.go

```diff
@@ -0,0 +1,72 @@
+// Copyright ©2022 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package r3
+
+import "math"
+
+// Triangle represents a triangle in 3D space and
+// is composed by 3 vectors corresponding to the position
+// of each of the vertices. Ordering of these vertices
+// decides the "normal" direction.
+type Triangle struct {
+	// Triangle vertices. Ordering of these vertices decides Normal vector direction.
+	// Inverting ordering of two vertices inverts the resulting direction.
+	V [3]Vec
+}
```

Comment:
just a drive-by comment:
any reason why we don't define it as:
```go
type Triangle [3]Vec
```

or, (better for the SSA pass):
```go
type Triangle struct {
    A, B, C Vec
}
```


do we expect to sub-slice `t.V`? or iterate over the vertices of a triangle?

Reply:
It's common to iterate over triangle vertices to make code more succinct. I can change it to `type Triangle [3]Vec`, which I prefer over the current implementation.